tests/models/test_model.py

```diff
@@ -224,11 +224,13 @@ def test_model_log_with_input_example_succeeds():
 
         # date column will get deserialized into string
         input_example["d"] = input_example["d"].apply(lambda x: x.isoformat())
-        pd.testing.assert_frame_equal(x, input_example)
+        pd.testing.assert_frame_equal(x, input_example.astype({"b": "string", "d": "string"}))
 
         loaded_example = loaded_model.load_input_example(local_path)
         assert isinstance(loaded_example, pd.DataFrame)
-        pd.testing.assert_frame_equal(loaded_example, input_example)
+        pd.testing.assert_frame_equal(
+            loaded_example, input_example.astype({"b": "string", "d": "string"})
+        )
```

Comment:
Did we discover some incompatibilities?

Reply:
We did. https://github.com/mlflow/mlflow/pull/6287/files#diff-d7b63e518f99708b094f8b946f0fc6dccfee5707a46f8892699ce0d39723ca7cR230-R232

In pandas < 1.4, `pandas.read_json` ignores non-numpy `dtype`:

```python
Python 3.7.10 (default, Jun  4 2021, 14:48:32) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas as pd
>>> pd.__version__
'1.3.1'
>>> j = '[{"col 1":"a","col 2":"b"},{"col 1":"c","col 2":"d"}]'
>>> pd.read_json(j, orient='records', dtype={"col 1": "string", "col 2": "string"}).dtypes
col 1    object
col 2    object
dtype: object

> python
Python 3.8.13 (default, Mar 28 2022, 11:38:47) 
[GCC 7.5.0] :: Anaconda, Inc. on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas as pd
>>> pd.__version__
'1.4.3'
>>> j = '[{"col 1":"a","col 2":"b"},{"col 1":"c","col 2":"d"}]'
>>> pd.read_json(j, orient='records', dtype={"col 1": "string", "col 2": "string"}).dtypes
col 1    string
col 2    string
dtype: object
```

Reply:
Very cool! :)